app/models/skin.rb

```diff
@@ -315,14 +325,16 @@ def get_all_parents
 
   # This is the main function that actually returns code to be embedded in a page
   def get_style(roles_to_include = DEFAULT_ROLES_TO_INCLUDE)
-    Rails.cache.fetch(skin_cache_html_key(self, roles_to_include)) do
-      style = ""
-      if self.get_role != "override" && self.get_role != "site"
-        style += AdminSetting.default_skin != Skin.default ? '' : (Skin.get_current_site_skin ? Skin.get_current_site_skin.get_style(roles_to_include) : '')
-      end
-      style += self.get_style_block(roles_to_include)
-      style.html_safe
+    style = ""
+
+    if self.get_role != "override" && self.get_role != "site" &&
+       self.id != AdminSetting.default_skin_id &&
+       AdminSetting.default_skin.is_a?(Skin)
+      style += AdminSetting.default_skin.get_style(roles_to_include)
```

Comment:
I'm not sure if using `AdminSetting.default_skin` here instead of `Skin.get_current_site_skin` is the right idea. Currently, `AdminSetting.default_skin = Skin.default`, which is a blank skin with the `user` role, so using `Skin.get_current_site_skin` is the only way to make the skins work. But if we run the following command before the deploy:
```
Skin.default.update(role: "site", skin_parents_attributes: [{ position: 1, parent_skin_title: "Archive 2.0" }])
```
Then the "Default" skin will be a site skin that has "Archive 2.0" listed as a parent, so the contents of the "Archive 2.0" skin will be included whenever the "Default" skin is used.

This has a couple of advantages: (1) the relationship between the "Archive 2.0" skin and the "Default" skin is explicit, which should make it easier to expire caches when `Skin.get_current_site_skin` changes versions or adds new files, and (2) it means that we don't have to check the file system to look up the most recent site skin version number every time a skin's cache expires.

I'm still not certain this is the right decision, though -- input welcome!

Reply:
The only potential problem I can think of is the Default skin not properly updating when Archive 2.0 changes. 

Right now, if I am using a skin that has parent skins, and I make a change to one of the parents, then I don't see those changes until I edit the one I'm actually using. 

_However_, it's possible the re-caching james_ does on deploy will mean that's a non-issue.

Reply:
The recaching I do on deploy is writing the files out to disk, and is nothing to do with memcached.

Reply:
Ah, I think that's because editing a skin calls `cache!` to reconstruct the file system cache on edit, but doesn't do the same for any child skins: https://github.com/otwcode/otwarchive/blob/54c7496ea8f717b0434ce4e550a5c0211649c985/app/controllers/skins_controller.rb#L93
I can fix that.

Reply:
That's [AO3-4551](https://otwarchive.atlassian.net/browse/AO3-4551), thanks for fixing it!

Reply:
> - As an admin, go to Skins > Approved Skins and pick a new skin to use as the default. Make sure that the new skin is included in subsequent page loads, both for logged out users, and for users with a custom skin adding onto the default.
> - As an admin, go to Skins > Approved Skins, click the link for the current default skin, and then edit it to add some CSS. Make sure that the changes are immediately reflected on the next page you visit, both for logged-out users, and for users with a custom skin adding onto the default.

And that's [AO3-4829](https://otwarchive.atlassian.net/browse/AO3-4829).

Reply:
> Then the "Default" skin will be a site skin that has "Archive 2.0" listed as a parent, so the contents of the "Archive 2.0" skin will be included whenever the "Default" skin is used.

With the old approach we need to test 2 cases:

- Admin default skin is `Skin.default`, include `Skin.get_current_site_skin`;
- Admin default skin is something else, include its `get_style` instead.

With the new approach we have just 1 test case:

- Admin default skin can be whatever, include its `get_style`.

I'm good with the new approach. Can you include the command to undo this if necessary?
```rb
Skin.default.update(role: "site", skin_parents_attributes: [{ position: 1, parent_skin_title: "Archive 2.0" }])
```

Reply:
The command to undo it is more complicated, but I believe this should do it:
```
Skin.transaction do
  skin = Skin.default
  skin.parent_skins = []
  skin.update(role: "user")
end
```